torch/utils/_pytree.py

```diff
@@ -304,7 +304,11 @@ def _broadcast_to_and_flatten(pytree: PyTree, spec: TreeSpec) -> Optional[List[A
     if _is_leaf(pytree):
         return [pytree] * spec.num_leaves
     if isinstance(spec, LeafSpec):
-        return None
+        if isinstance(pytree, (list, tuple)):
```

Comment:
Should be more general? What if it is a `dict`, `OrderedDict` or `namedtuple` in `SUPPORTED_NODES`